app/backend/core/authentication.py

```diff
@@ -57,11 +59,14 @@ def __init__(
                 client_credential=server_app_secret,
                 token_cache=PersistedTokenCache(persistence),
             )
+        else:
```

Comment:
Related to line 52, not this line: when deployed, App Service always logs the exception for me of "Encryption unavailable. Opting in to plain text." 
When I looked into that, I see the msal-extensions docs seem to indicate that it shouldn't be used on deployed machines-

https://pypi.org/project/msal-extensions/

"Note: It is recommended to use this library for cache persistance support for Public client applications such as Desktop apps only. In web applications, this may lead to scale and performance issues. Web applications are recommended to persist the cache in session. Take a look at this [webapp sample](https://github.com/Azure-Samples/ms-identity-python-webapp)."

Is it expected that it logs the exception in prod and uses FilePersistence instead? If its expected, should we change it to an info/warning? It adds a lot of noise to logs since it includes the full traceback.

Reply:
We don’t use sessions in our application. We can add support for this in app.py and remove the file persistence cache completely.

the encrypted one only works if you install g object which I found difficult

Reply:
Discussed offline - replace with secure cookie sessions. Requires key vault support